### DISCUSSION TITLE
STM32H743 is not recognized as MSC

### DISCUSSION THREAD

**Discussion by Sophi** [micropython/micropython#15856] [STM32 / Pyboard] [answered]
Hello! 
I am new to the STM32 family and have a custom board which I am trying to load a *.py onto.
I have built and loaded the MicroPython firmware for this chip (STM32H743) using NUCLEO_H743ZI as a base.
The only file I changed was mpconfigboard.h.

I can load the bin, elf, or dfn using DFN or SWD w STLink pins. I have access to boot and reset pins (can pull high or low).
I cannot see the board as a MSC to load *py files onto. 

I must be missing something, perhaps in the *.h or in the boot sequence, but have not been able to find this info anywhere.
Any ideas? 

Thanks in advance!

**Comment by mattytrentini** [comment, marked as answer]
Most MicroPython boards, including the H743ZI, don't mount as a mass storage device on boot. You'll typically interact with them with a tool such as [mpremote](https://github.com/micropython/micropython/tree/master/tools/mpremote) that will allow you to access the REPL and transfer files. Hope that helps!

**Comment by Sophi** [comment]
That's very helpful, thank you!

However, when I use mpremote (or screen), the board isn't recognized. 
I can see the USB to UART cable blinking and it shows up in my system info (Mac). 

The board shows up there as DFU in FS mode. 
<img width="414" alt="Screenshot 2024-09-17 at 10 22 43 AM" src="https://github.com/user-attachments/assets/34d6f7f5-d0f4-41c3-bf45-ebeca9dc0969">

I have NRST pulled high and Boot0 low.

**Reply by ned-pcs** [reply]
`mpremote` is for working with a board once you have put MicroPython onto it.
You need to first transfer your binary to the board; this can be done using `st-flash` or similar tool. See https://micropython.org/download/NUCLEO_H743ZI/ for instructions.

**Comment by andrewleech** [comment]
Boot0 high is normally the trigger for running the rom bootloader you're seeing there.  [edited]

You can use that DFU mode for loading micropython firmware if needed, else if it's already flashed you should be able to run micropython with Boot0 high. 

If that still doesn't appear to run directly you might need to look closer at the crystal / HSE clock configuration settings for the build. Many nucleo boards do not use a crystal for the main clock, instead using a clock signal provided by the on-board stlink. If your custom board uses a crystal then this may need configuring in the board profile.

**Comment by Sophi** [comment]
Thanks all! 
Confirming that to see the board in REPL or as a MSC Boot0 should be high? 

I will take a look at the clock settings in mpconfigboard.h as I have an external crystal.

**Reply by andrewleech** [reply]
Sorry I just checked again / my mistake - Boot0 should be low to boot normal, Boot0 high should be to get to builtin bootloader.
I'm not sure now how it's going to bootloader with Boot0 low.
